container-disc/src/test/java/com/yahoo/container/jdisc/metric/GarbageCollectionMetricsTest.java

```diff
@@ -0,0 +1,60 @@
+package com.yahoo.container.jdisc.metric;
+
+import com.yahoo.jdisc.Metric;
+import org.junit.Test;
+
+import java.lang.management.ManagementFactory;
+import java.time.Clock;
+import java.util.LinkedList;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.atLeast;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author ollivir
+ */
+public class GarbageCollectionMetricsTest {
+    @Test
+    public void gc_metrics_reported_at_intervals() {
```

Comment:
I suggest to express test names as requirements, e.g. `gc_metrics_are_reported_at_regular_intervals`. Then, we never need to guess what the test is trying to verify.

Reply:
The name was also incorrect and based on how the logic worked in an earlier version. Now renamed to `gc_metrics_are_collected_in_a_sliding_window`